src/internal/bolt/bolt-protocol-v1.js

```diff
@@ -91,16 +109,13 @@ export default class BoltProtocol {
    */
   initialize ({ userAgent, authToken, onError, onComplete } = {}) {
     const observer = new LoginObserver({
-      connection: this._connection,
+      beforeError: err => this._onProtocolError(err.message),
```

Comment:
Or we could go with `onError`. In the original LoginObserver, it did something like:

```
if(this.beforeError)
   this.beforeError(error)
// change some state on the connection
if (this.afterError)
  this.afterError(error)
```

Since we removed the part on the middle, i think the methods should join together as onError and then the initialize method should handle the call to the this._onProtocolError and redirect the on error call.